### PR TITLE
backupccl: add random data to tables in TestBackupRestoreDataRoundtrips

### LINKED ISSUE #72345: bulkio: follow through on random testing
[This PR](https://github.com/cockroachdb/cockroach/pull/71627) introduced welcome randomness to BACKUP/RESTORE. A couple of follow-through ideas:

- Add some random data via SQLSmith (per code comment)
- Move this into another package per @dt's suggestion in CRL Slack. Requires some refactoring.

FYI @jordanlewis 

Epic CRDB-10683

Jira issue: CRDB-11103

### PR DESCRIPTION
**Pull request by msbutler**
Previously, TestBackupRestoreDataRoundtrips did not test if data in the
randomly created tables were preserved after a backup restore roundtrip.
This patch tries to add random data to each table and tests that the data is
preserved across the roundtrip. In addition, this pr moves this test
to its own package because the test takes a long time to run.

Fixes #72345

Release justification: changes to test
Release note: None

### CODE REVIEW

**Review by jordanlewis — COMMENTED** [review]
Cool! Looks good, but I'll leave the review to your team :)

File: pkg/ccl/backupccl/backuprand/main_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 The Cockroach Authors.
```
Comment (stevendanna) [inline pkg/ccl/backupccl/backuprand/main_test.go:1]: [nit] 2022

File: pkg/ccl/backupccl/backuputils/testutils.go [hunk 2]
```diff
@@ -114,3 +118,60 @@ func BackupRestoreTestSetup(
...
+		`SELECT job_id, status, fraction_completed FROM crdb_internal.jobs WHERE job_id = $1`, actualJob.id,
+	).Scan(
+		&expectedJob.id, &expectedJob.status, &expectedJob.fractionCompleted,
+	)
+
+	if e, a := expectedJob, actualJob; !reflect.DeepEqual(e, a) {
+		return errors.Errorf("result does not match system.jobs:\n%s",
+			strings.Join(pretty.Diff(e, a), "\n"))
+	}
+
```
Comment (stevendanna) [inline pkg/ccl/backupccl/backuputils/testutils.go:175]: If you felt like it, I think `require.Equal` provides pretty nice output by default and could simplify this function a bit.

**Review by stevendanna — APPROVED** [review]
Apologies this sat for so long. LGTM.

File: pkg/ccl/backupccl/backuprand/main_test.go (hunk 1 above)
Comment (msbutler) [reply pkg/ccl/backupccl/backuprand/main_test.go]: Done.

File: pkg/ccl/backupccl/backuputils/testutils.go (hunk 2 above)
Comment (msbutler) [reply pkg/ccl/backupccl/backuputils/testutils.go]: Done.

**Comment by msbutler** [comment]
bors r=stevendanna

**Pull request merged**